numeric.c

```diff
@@ -3280,9 +3280,6 @@ int_even_p(VALUE num)
     else if (RB_TYPE_P(num, T_BIGNUM)) {
```

Comment:
Maybe changing this line to `else {` and asserting `RB_TYPE_P(num, T_BIGNUM)` in it might be better to double check your assumption. And then `return Qfalse` had better live in `if (FIXNUM_P(num))` block as it's no longer shared.